File: 0204-count-primes/0204-count-primes.cs
public class Solution {
    public int CountPrimes(int n) {
        List<int> v = new List<int>(new int[n + 1]);
        for (int i = 0; i <= n; i++) {
            v[i] = 1;  // Initialize the list with 1
        }

        int cnt = 0;
        for (int i = 2; i <= Math.Sqrt(n); i++) {
            if (v[i] == 1) {
                for (int j = i * i; j < n; j += i) {
                    v[j] = 0;  // Mark non-prime numbers as 0
                }
            }
        }

        for (int i = 2; i < n; i++) {
            if (v[i] == 1) {
                cnt++;
            }
        }

        return cnt;
    }
}

File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<int> v(n+1,1);
        int cnt = 0;
        for(int i=2;i<=sqrt(n);i++){
            if(v[i]){
                for(int j =i*i; j<n; j+=i){
                    v[j] = 0;
                }
            }
        }
        for(int i=2;i<n;i++){
            if(v[i]) cnt++;
        }
        return cnt;
    }
};